eureka2-core/src/main/java/com/netflix/eureka2/connection/RetryableConnectionFactory.java

```diff
@@ -14,50 +14,55 @@
 import rx.functions.Func2;
 import rx.subjects.AsyncSubject;
 import rx.subjects.BehaviorSubject;
+import rx.subjects.Subject;
 
 import java.util.concurrent.atomic.AtomicBoolean;
 
 /**
- * Abstract factory that provides a {@link RetryableConnection} for each newConnection call.
+ * Abstract factory that provides a {@link RetryableConnection} for each unaryConnection call.
  *
  * @param <CHANNEL> the type of channel to be used
- * @param <OP> the type of op to be applied to the channel
  *
  * @author David Liu
  */
-public abstract class RetryableConnectionFactory<CHANNEL extends ServiceChannel, OP> {
+public class RetryableConnectionFactory<CHANNEL extends ServiceChannel> {
 
     private static final Logger logger = LoggerFactory.getLogger(RetryableConnectionFactory.class);
 
-    private final ChannelFactory<CHANNEL> channelFactory;
+    protected final ChannelFactory<CHANNEL> channelFactory;
 
     public RetryableConnectionFactory(final ChannelFactory<CHANNEL> channelFactory) {
         this.channelFactory = channelFactory;
     }
 
     /**
+     * @param executeOnChannel a func1 that describes a nullary call for the channel
+     * @return a {@link RetryableConnection} that contains several observables. This lifecycle observable provided
+     * can be retried on.
+     */
+    public RetryableConnection<CHANNEL> nullaryConnection(final Func1<CHANNEL, Observable<Void>> executeOnChannel) {
```

Comment:
I understand that this is type of connection where a client does not provide any operations. Can we pickup a better name for it, and also for unaryConnection? Some documentation would be useful, as well.